services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +42,52 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if (&thermalMgrQueueHandle != NULL) {
+    if (event != NULL) {
+      if (xQueueSend(thermalMgrQueueHandle, (void *) event, (TickType_t) 0) == pdPASS)
+        return ERR_CODE_SUCCESS;
+      else
+        return ERR_CODE_QUEUE_FULL;
+    }
+    else
+      return ERR_CODE_INVALID_ARG;
+  }
 
-  return ERR_CODE_SUCCESS;
+  return ERR_CODE_INVALID_QUEUE_MSG;
```

Comment:
If the thermal mgr queue handle is null, then it makes more sense to return ERR_CODE_INVALID_STATE instead since we are in a state where we have not intitialized the queue yet and it's not related to the queue msg itself